components/tools/OmeroPy/src/omero/plugins/ldap.py

```diff
@@ -163,7 +166,15 @@ def getdn(self, args):
         if dn is not None and dn.strip():
             self.ctx.out(dn)
 
-    @admin_only
+    @admin_only(AdminPrivilegeModifyGroup)
+    def update_group(self, iupdate, group):
+        iupdate.saveObject(group)
+
+    @admin_only(AdminPrivilegeModifyUser)
+    def update_user(self, iupdate, user):
+        iupdate.saveObject(user)
+
+    @admin_only()
```

Comment:
I thought we were keen to use the decorator though? Otherwise (renamed) `_check_admin` becomes more API we commit to offering to consumers. Can make this change though if desired. I also wonder if there's an easy way to use `admin_only` inline, maybe not.